java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java

```diff
@@ -1028,4 +1036,112 @@ public void testRemoveSpecificAPIpatterns() throws IOException {
             "}")
         .doTest();
   }
+
+  public void transformAndCreateNewPropertyFile(
+      String srcProp, String trgtProp, String stale_flag, boolean isTreated) throws IOException {
+
+    String[] temp = stale_flag.split("_");
+    String methodPropertiesToAdd =
+        String.join(
+            "\n",
+            "{",
+            "      \"methodName\": \"$flagName$.getValue\",",
+            "      \"flagType\": \"treated\",",
+            "      \"returnType\": \"boolean\"",
+            "    },",
+            "    {",
+            "      \"methodName\": \"put\",",
+            "      \"flagType\": \"empty\",",
+            "      \"argumentIndex\": 1",
+            "    },",
+            "");
+    String annotationPropertiesToAdd =
+        String.join(
+            "\n",
+            "{",
+            "      \"name\" : \"BoolParam\",",
+            "      \"flag\" : \"$key\",",
+            "      \"treated\" : \"$notIsTreated$\"",
+            "    },",
+            "{",
+            "        \"name\" : \"PVal\",",
+            "        \"flag\" : \"$key\",",
+            "        \"treated\" : \"$val\"",
+            "    }");
+    String flagNameCamelCase =
+        temp[0]
+            + Arrays.stream(temp, 1, temp.length)
+                .map(str -> str.substring(0, 1).toUpperCase() + str.substring(1))
+                .collect(joining());
+    String newContent =
+        Files.readAllLines(Paths.get(srcProp))
+            .stream()
+            .map(
+                x ->
+                    x.replace(
+                        "\"methodProperties\": [",
+                        "\"methodProperties\": [" + methodPropertiesToAdd))
+            .map(
+                x ->
+                    x.replace(
+                        "\"annotations\": [", "\"annotations\": [" + annotationPropertiesToAdd))
+            .map(x -> x.replace("$flagName$", flagNameCamelCase))
+            .map(x -> x.replace("$notIsTreated$", String.valueOf(!isTreated)))
+            .collect(joining("\n"));
+    Path trgt = Paths.get(trgtProp);
+    Files.deleteIfExists(trgt);
+    Files.write(trgt, newContent.getBytes(UTF_8), StandardOpenOption.CREATE);
+  }
+
+  @Test
+  public void testMethodChainTreated() throws IOException {
+    String stale_flag = "stale_flag";
+    String isTreated = "true";
+    String srcProp = "config/properties.json";
```

Comment:
this can read from a different properties.json with the $....$ parameters in there.